code/script.js

```diff
@@ -225,89 +236,124 @@ const generateBoard = () => {
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
-}
-
-// This function to start (and restart) the game
-const start = () => {
-  // Here we're setting charactersInPlay array to be all the characters to start with
-  charactersInPlay = CHARACTERS
-  // What else should happen when we start the game?
+  console.log(secret)
 }
 
 // setting the currentQuestion object when you select something in the dropdown
-const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
   // const value =
+const selectQuestion = () => {
+  const value = questions.value //referring to the value in the optgroups in the index-file
+  const category = questions.options[questions.selectedIndex].parentNode.label 
 
-  currentQuestion = {
-    category: category,
-    // value: value
-  }
+  if (category === 'hair') {
+    currentQuestion = {
+      category: 'hair',
+      value: value
+    }
+  } else if (category === 'eyes') {
+    currentQuestion = {
+      category: 'eyes',
+      value: value
+    }
+  } else if (category === 'accessories') {
+    currentQuestion = {
+      category: 'accessories',
+      value: value
+    }
+  } else {
+    currentQuestion = {
+      category: 'other',
+      value: value
+    }
+  }   
 }
 
-// This function should be invoked when you click on 'Find Out' button.
-const checkQuestion = () => {
-  const { category, value } = currentQuestion
 
-  // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
-  // See if we should keep or remove people based on that
-  // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
+// Compares the currentQuestion with the secret person based on category
+// Eyes and hair can be compared by compairing the value of the property string from the secret person and all other characters on the board
+// Accessories and other can be compared by caompairing the arrays and therefor we use .includes(value)
+const checkQuestion = () => {
+  const { category, value } = currentQuestion; 
 
-  } else if (category === 'accessories' || category === 'other') {
+  if (category === 'eyes' || category === 'hair') {
+    if (value === secret.eyes || value === secret.hair) {
+      filterCharacters(true);
+    } else {
+      filterCharacters();
+    }  } 
 
+  else {
+    if (secret.accessories.includes(value) || secret.other.includes(value)) {
+      filterCharacters(true);
+    } else {
+      filterCharacters();
+    }
   }
+  counter++; 
```

Comment:
Nice that you made this work!!! 👊🏻👌🏻 Is the counter a variable? Or how does it work?